pkg/datagatherer/k8s/generic.go

```diff
@@ -110,6 +111,12 @@ func (g *DataGatherer) Fetch() (interface{}, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	// Redact Secret data
+	if strings.ToLower(g.groupVersionResource.Resource) == "secret" {
```

Comment:
I think you also need to match `secrets` here too, as technically `secrets` with an 's' is the *resource name* (as part of the REST path).

I imagine there's also a clever way we can use the `runtime.Scheme` to inspect the list type to check if it == `SecretList` (i.e. the resource *kind*), which is more concrete.

Reply:
Yeah, changing to `secrets` with an 's' has fixed things!

I did try doing `if list.Object["kind"] == "SecretList" {` here, however while that worked against a real cluster is didn't work with the test suite.

Reply:
So I think a better approach is to use a runtime.Scheme to *identify* the type of resource, instead of relying on things like the `kind` or `apiVersion` being set. A runtime.Scheme will be able to use the type information registered with it to identify what kind it is. Let me have a dig now and see what method there is for this...

Reply:
https://godoc.org/k8s.io/apimachinery/pkg/runtime#Scheme.ObjectKinds this should do it 😄 from there you can iterate over all the returned GVKs and if any of them have a `kind` of `Secret` and a `group` of either `core` or `` (empty string) then redact it

Reply:
Ah. you'll actually need to match on `SecretList` come to think.. but same difference 😄

Reply:
You will need to use a 'scheme' that has the 'Secret' and 'SecretList' kind registered with it - Kubernetes provides this as `k8s.io/client-go/kubernetes/scheme.Scheme`, which you should be safe to directly reference here